src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -182,10 +193,10 @@ protected static QueryParseContext createContext() {
         return new QueryParseContext(index, queryParserService);
     }
 
-    private static void assertQueryHeader(XContentParser parser, String expectedParserName) throws IOException {
+    private void assertQueryHeader(XContentParser parser) throws IOException {
         assertThat(parser.nextToken(), is(XContentParser.Token.START_OBJECT));
         assertThat(parser.nextToken(), is(XContentParser.Token.FIELD_NAME));
-        assertThat(parser.currentName(), is(expectedParserName));
+        assertThat(parser.currentName(), is(testQuery.parserName()));
```

Comment:
not sure either, I just thought we introduced `parserName()` to have our temporary `toQuery()` method working.

```
//norelease to be removed once all query builders override toQuery providing their own specific implementation.
    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
        return parseContext.indexQueryParserService().queryParser(parserName()).parse(parseContext);
    }
```